pages/404.tsx

```diff
@@ -1,27 +1,30 @@
 import Link from "next/link";
 import { useEffect, useState } from "react";
-import { differenceInDays } from "date-fns";
 import FullHeightContainer from "components/atoms/FullHeightContainer/full-height-container";
 import HeaderLogo from "components/molecules/HeaderLogo/header-logo";
 import { useFetchTopContributors } from "lib/hooks/useFetchTopContributors";
 import DevCardWall from "components/organisms/DevCardWall/dev-card-wall";
-import { DevCardProps } from "components/molecules/DevCard/dev-card";
-import { fetchContributorPRs } from "lib/hooks/api/useContributorPullRequests";
-import { getRepoList } from "lib/hooks/useRepoList";
-import getContributorPullRequestVelocity from "lib/utils/get-contributor-pr-velocity";
-import getPercent from "lib/utils/get-percent";
-import { getAvatarByUsername } from "lib/utils/github";
 import BubbleBG from "../img/bubble-bg.svg";
+import { UserDevStats } from "./u/[username]/card";
+
+async function fetchUserData(username: string) {
+  const response = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/users/${username}/devstats`, {
+    headers: {
+      "Content-Type": "application/json"
+    }
+  });
+  return await response.json() as UserDevStats;
+};
 
 export default function Custom404() {
   const { data } = useFetchTopContributors({ limit: 20 });
-  const [cards, setCards] = useState<DevCardProps[]>([]);
+  const [cards, setCards] = useState<UserDevStats[]>([]);
   const [isLoading, setIsLoading] = useState<boolean>(true);
   const [initialCardIndex, setInitialCardIndex] = useState<number | undefined>();
 
   useEffect(() => {
     async function loadCards() {
-      const cardData = await Promise.all(data.map((user) => getAllCardData(user.login)));
+      const cardData = await Promise.all(data.map((user) => fetchUserData(user.login)));
```

Comment:
Not for this PR, but it might be good to have a call that gets the card data for all the users in one shot. Not sure how many you're requesting atm.